openmmtools/mcmc.py

```diff
@@ -146,9 +146,19 @@ class MCMCMove(SubhookedABCMeta):
     statistics such as number of attempted moves and acceptance rates.
 
     """
+    def __init__(self, context_cache=None):
+        if context_cache is not None:
+            logger.warning("Ignoring context_cache argument. Specifying context_cache on" 
+                           "initialization of an MCMCMove is deprecated; please pass to"
+                           " apply() method as kwarg instead.")
+        # hardcoding unused attribute to global context cache. For compatibility.
+        self.context_cache = cache.global_context_cache
```

Comment:
Is this only to support subclasses of `MCMCMove` that may reference `self.context_cache`? 
If so, I wonder if we want to make this a `@property` so we can also issue the warning when someone tries to access `self.context_cache`?
```python
@property
def context_cache(self):
    logger.warning('The MCMCMove.context_cache field has been deprecated. The API now requires context_cache be passed to apply(). Please update your code.')
    from openmmtools import cache
    return cache.global_context_cache
```

Reply:
Should be fixed now in latest version.